scene/gui/line_edit.cpp

```diff
@@ -633,8 +633,8 @@ void LineEdit::_notification(int p_what) {
 				if (char_ofs >= t.length())
 					break;
 
-				CharType cchar = pass ? '*' : t[char_ofs];
-				CharType next = pass ? '*' : t[char_ofs + 1];
+				CharType cchar = (text.empty() || !pass) ? t[char_ofs] : '*';
```

Comment:
Keeping the original order and testing `(pass && !text.empty())` would actually be more efficient IINM, as the second condition would only be tried if `pass` is true.